python/lancedb/rerankers/linear_combination.py

```diff
@@ -0,0 +1,111 @@
+        with the relevance score.
+    """
+
+    def __init__(
+        self, weight: float = 0.7, fill: float = 1.0, return_score="relevance"
+    ):
+        if weight < 0 or weight > 1:
+            raise ValueError("weight must be between 0 and 1.")
+        super().__init__(return_score)
+        self.weight = weight
+        self.fill = fill
+
+    def rerank_hybrid(
+        self,
+        query_builder: "lancedb.HybridQueryBuilder",  # noqa: F821
+        vector_results: pa.Table,
+        fts_results: pa.Table,
+    ):
+        combined_results = self.merge_results(vector_results, fts_results, self.fill)
+
+        return combined_results
+
+    def merge_results(
+        self, vector_results: pa.Table, fts_results: pa.Table, fill: float
+    ):
+        # If both are empty then just return an empty table
+        if len(vector_results) == 0 and len(fts_results) == 0:
+            return vector_results
+        # If one is empty then return the other
+        if len(vector_results) == 0:
+            return fts_results
+        if len(fts_results) == 0:
+            return vector_results
+
+        # sort both input tables on _rowid
+        combined_list = []
+        vector_list = vector_results.sort_by("_rowid").to_pylist()
+        fts_list = fts_results.sort_by("_rowid").to_pylist()
+        i, j = 0, 0
+        while i < len(vector_list):
+            if j >= len(fts_list):
+                for vi in vector_list[i:]:
+                    vi["_relevance_score"] = self._combine_score(vi["_distance"], fill)
+                    combined_list.append(vi)
+                break
+
+            vi = vector_list[i]
+            fj = fts_list[j]
+            # invert the fts score
+            inverted_fts_score = 1 - fj["score"]
+            if vi["_rowid"] == fj["_rowid"]:
+                vi["_relevance_score"] = self._combine_score(
+                    vi["_distance"], inverted_fts_score
+                )
+                vi["score"] = fj["score"]  # keep the original score
+                combined_list.append(vi)
+                i += 1
+                j += 1
+            elif vector_list[i]["_rowid"] < fts_list[j]["_rowid"]:
+                vi["_relevance_score"] = self._combine_score(vi["_distance"], fill)
+                combined_list.append(vi)
+                i += 1
+            else:
+                fj["_relevance_score"] = self._combine_score(inverted_fts_score, fill)
+                combined_list.append(fj)
+                j += 1
+        if j < len(fts_list) - 1:
+            for fj in fts_list[j:]:
+                fj["_relevance_score"] = self._combine_score(inverted_fts_score, fill)
+                combined_list.append(fj)
+
+        relevance_score_schema = pa.schema(
+            [
+                pa.field("_relevance_score", pa.float32()),
+            ]
+        )
+        combined_schema = pa.unify_schemas(
+            [vector_results.schema, fts_results.schema, relevance_score_schema]
```

Comment:
Not sure if this is a decent/optimal way to merge schema

Reply:
Is there a run time penalty you observed as a result of this? The overall algorithm seemed to be quite fast from what you showed.